src/components/textfield/TextField.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './textfield.scss';
+
+class TextField extends React.Component {
+  render() {
+    const { value, onChange, onFocus, onBlur, label, placeholder, type, multiline } = this.props;
+
+    const _renderTextInput = () => {
+      if (!multiline) {
+        return (
+          <div className={styles.root}>
+            <label className={styles.labelStyle}>{label}</label>
+            <input
+              className={styles.inputStyle}
+              label={label}
+              onChange={onChange}
+              onBlur={onBlur}
+              onFocus={onFocus}
+              type={type}
+              value={value}
+              placeholder={placeholder}
+            />
+          </div>
+        );
+      } else {
+        return (
+          <div className={styles.root}>
+            <label className={styles.labelStyle}>{label}</label>
+            <textarea
+              className={styles.textAreaStyle}
+              label={label}
+              onChange={onChange}
+              onBlur={onBlur}
+              onFocus={onFocus}
+              type={type}
+              value={value}
+              placeholder={placeholder}
+            />
+          </div>
+        );
+      }
+    };
+    return _renderTextInput();
+  }
+}
+
+TextField.propTypes = {
+  onChange: PropTypes.func,
+  value: PropTypes.string,
+  onBlur: PropTypes.func,
+  onFocus: PropTypes.func,
+  placeholder: PropTypes.string,
+  type: PropTypes.string,
```

Comment:
I think you should use `oneOf`, like this: `type: oneOf(['text', 'password', 'email', 'etc...']),`

Here's a list of types:

color
date
datetime-local
email
month
number
range
search
tel
time
url
week

Also, don't forget to set the default value of `type` to `text`.

Reply:
+1